packages/blocks/src/_common/theme/theme-observer.ts

```diff
@@ -1,45 +1,32 @@
 import { Slot } from '@blocksuite/global/utils';
 
 import type { Color } from '../../surface-block/consts.js';
-import type { CssVariablesMap } from './css-variables.js';
-
-import { StyleVariables, isCssVariable } from './css-variables.js';
 
 export enum ColorScheme {
   Dark = 'dark',
   Light = 'light',
-  Normal = 'normal',
-}
-
-export function extractCssVariables(element: Element): CssVariablesMap {
-  const styles = window.getComputedStyle(element);
-  const variables = StyleVariables.reduce((acc, cssName) => {
-    const value = styles.getPropertyValue(cssName).trim();
-    acc[cssName] = value;
-
-    // --affine-palette-transparent: special values added for the sake of logical consistency.
-    if (cssName === '--affine-palette-transparent' && !value) {
-      acc[cssName] = '#00000000';
-    }
-
-    return acc;
-  }, {} as CssVariablesMap);
-  return variables;
 }
 
 /**
  * Observer theme changing by `data-theme` property
  */
-export class ThemeObserver extends Slot<CssVariablesMap> {
-  private _cssVariables: CssVariablesMap | null = null;
+export class ThemeObserver extends Slot<ColorScheme> {
+  static #computedStyle: CSSStyleDeclaration;
 
-  private _mode: ColorScheme = ColorScheme.Normal;
+  static #instance: ThemeObserver;
 
-  private _observer?: MutationObserver;
+  #mode: ColorScheme = ColorScheme.Light;
```

Comment:
What if just make `mode` a signal?

Reply:
I've thought about this too, it should be fine.